frontend/lib/Logging/LoginView.dart

```diff
@@ -0,0 +1,331 @@
+      setState(() {
+        _formType = FormType.values[_tabController.index];
+      });
+    });
+  }
+
+  @override
+  void dispose() {
+    _tabController.dispose();
+    super.dispose();
+  }
+
+  void enableSubmitButton(ButtonState buttonState) {
+    setState(() {
+      if (_formType == FormType.signin) {
+        _loginButtonState = buttonState;
+      } else if (_formType == FormType.signup) {
+        _createButtonState = buttonState;
+      }
+    });
+  }
+
+  void _listenForLoginInput() {
+    if (_emailLoginController.text.isNotEmpty &&
+        _passwordLoginController.text.isNotEmpty &&
+        checkEmailRegex(_emailLoginController.text)) {
+      enableSubmitButton(ButtonState.enabled);
+    } else {
+      enableSubmitButton(ButtonState.disabled);
+    }
+  }
+
+  void _listenForCreateInput() {
+    if (_emailCreateController.text.isNotEmpty &&
+        _passwordCreateController.text.isNotEmpty &&
+        checkEmailRegex(_emailCreateController.text) &&
+        checkPasswordRegex(_passwordCreateController.text)) {
+      enableSubmitButton(ButtonState.enabled);
+    } else {
+      enableSubmitButton(ButtonState.disabled);
+    }
+  }
+
+  _LoginViewState() {
+    _emailLoginController.addListener(_listenForLoginInput);
+    _passwordLoginController.addListener(_listenForLoginInput);
+    _emailCreateController.addListener(_listenForCreateInput);
+    _passwordCreateController.addListener(_listenForCreateInput);
+  }
+
+  void onSubmitButton() {
+    late AuthenticateForm form;
+    if (_formType == FormType.signin) {
+      form = AuthenticateForm(_emailLoginController.text,
+          _passwordLoginController.text, FormType.signin);
+    } else if (_formType == FormType.signup) {
+      form = AuthenticateForm(_emailCreateController.text,
+          _passwordLoginController.text, FormType.signup);
+    }
+  }
+
+  Container _buildAppBar() => Container(
+        height: 200,
+        padding: const EdgeInsets.only(top: 10, left: 10, right: 10),
+        decoration: const BoxDecoration(color: Colors.transparent),
+        child: const Text('AIR POLLUTION APP',
+            style: TextStyle(
+                fontFamily: 'SofiaSans', fontSize: 55, color: Colors.white)),
+      );
+
+  Container _buildEmailInput(TextEditingController controller) => Container(
+      decoration: backgroundDecor,
+      alignment: Alignment.center,
+      padding: EdgeInsets.only(
+          left: MediaQuery.of(context).size.width * 0.1 - 10,
+          right: MediaQuery.of(context).size.width * 0.1 - 10,
+          top: 40,
+          bottom: 25),
+      child: Theme(
+        data: Theme.of(context).copyWith(colorScheme: loginPageTheme),
+        child: TextField(
+          controller: controller,
+          decoration: const InputDecoration(
+            focusedBorder: OutlineInputBorder(
+                borderSide:
+                    BorderSide(width: 2.5, color: loginPagePrimaryColor)),
+            labelText: 'Login',
```

Comment:
Z tego co widzę, to wybraliśmy `email` jako formę logowania. Myślę, że powinniśmy zmienić tutaj nazwę na `email`, będzie bardziej adekwatna do tego, co tam trzeba wpisać

Reply:
Zapomniałam, do zmiany